crates/agglayer-storage/src/storage/backup/mod.rs

```diff
@@ -0,0 +1,327 @@
+
+        let mut engine = RocksBackupEngine::open(&opts, &env)?;
+
+        std::fs::create_dir_all(db_path).unwrap();
+
+        Ok(engine.restore_from_backup(db_path, db_path, &RestoreOptions::default(), version)?)
+    }
+
+    pub fn list_backups(path: &Path) -> Result<BackupReport, BackupError> {
+        let env = rocksdb::Env::new()?;
+
+        let mut report = BackupReport::default();
+
+        let opts = BackupEngineOptions::new(path.join("state"))?;
+        let engine = RocksBackupEngine::open(&opts, &env)?;
+
+        report.state(engine.get_backup_info());
+
+        let opts = BackupEngineOptions::new(path.join("pending"))?;
+        let engine = RocksBackupEngine::open(&opts, &env)?;
+
+        report.pending(engine.get_backup_info());
+
+        let epoch_path = path.join("epochs");
+        let mut epochs = (read_dir(&epoch_path)?)
+            .flatten()
+            .filter_map(|d| {
+                d.file_name()
+                    .to_string_lossy()
+                    .parse::<u64>()
+                    .map(|r| (r, d.path()))
+                    .ok()
+            })
+            .collect::<Vec<_>>();
+
+        epochs.sort_by(|(p, _), (n, _)| p.cmp(n));
+
+        for (epoch_number, path) in epochs {
+            let opts = BackupEngineOptions::new(path)?;
+            let engine = RocksBackupEngine::open(&opts, &env)?;
+            report.epochs(epoch_number, engine.get_backup_info());
+        }
+
+        Ok(report)
+    }
+}
+
+impl Drop for BackupEngine {
+    fn drop(&mut self) {
+        self.env.set_background_threads(0);
+        self.env.set_low_priority_background_threads(0);
+        self.env.set_high_priority_background_threads(0);
+        self.env.set_bottom_priority_background_threads(0);
+
+        self.env.join_all_threads();
+    }
+}
+
+#[derive(Default, Serialize, Deserialize)]
+pub struct BackupReport {
+    epochs: BTreeMap<u64, Vec<BackupEngineInfo>>,
+    state: Vec<BackupEngineInfo>,
+    pending: Vec<BackupEngineInfo>,
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct BackupEngineInfo {
+    pub backup_id: u32,
+    #[serde(serialize_with = "timestamp_to_readable")]
+    pub timestamp: i64,
+    pub size: u64,
+    pub num_files: u32,
+}
+
+fn timestamp_to_readable<S>(timestamp: &i64, serializer: S) -> Result<S::Ok, S::Error>
+where
+    S: serde::Serializer,
+{
+    chrono::DateTime::<chrono::Utc>::from_timestamp(*timestamp, 0)
+        .unwrap_or_default()
+        .to_rfc2822()
+        .serialize(serializer)
+}
+
+impl From<RocksBackupEngineInfo> for BackupEngineInfo {
+    fn from(info: RocksBackupEngineInfo) -> Self {
+        Self {
+            backup_id: info.backup_id,
+            timestamp: info.timestamp,
+            size: info.size,
+            num_files: info.num_files,
+        }
+    }
+}
+
+impl BackupReport {
+    pub fn state<V: IntoIterator<Item = T>, T: Into<BackupEngineInfo>>(&mut self, value: V) {
```

Comment:
Nit: I think `set_state` (and `set_pending`, `set_epochs`) would be more appropriate. This way it feels like a getter.